openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -517,3 +521,37 @@ def _validate_email(email):
         raise AccountEmailInvalid(
             u"Email '{email}' format is not valid".format(email=email)
         )
+
+def set_has_profile_image(username, has_profile_image=True):
```

Comment:
I was forgetting that delete will go through this... Don't special-case based on the value of has_profile_image.

So you should pass in the requesting user, but then allow username != request.user.username if request.user.is_staff is true.